Description: Proper unescaping in links
Author: Carsten Dominik <carsten.dominik@gmail.com>
Origin: upstream
Bug-Debian: http://bugs.debian.org/572404

Index: org-mode-6.34c/lisp/org.el
===================================================================
--- org-mode-6.34c.orig/lisp/org.el	2010-01-18 00:10:45.000000000 +0100
+++ org-mode-6.34c/lisp/org.el	2010-03-05 11:27:52.000000000 +0100
@@ -7914,12 +7914,14 @@
       (url-unhex-string text)
     (setq table (or table org-link-escape-chars))
     (when text
-      (let ((re (mapconcat (lambda (x) (regexp-quote (cdr x)))
+      (let ((case-fold-search t)
+	    (re (mapconcat (lambda (x) (regexp-quote (downcase (cdr x))))
 			   table "\\|")))
 	(while (string-match re text)
 	  (setq text
 		(replace-match
-		 (char-to-string (car (rassoc (match-string 0 text) table)))
+		 (char-to-string (car (rassoc (upcase (match-string 0 text))
+					      table)))
 		 t t text)))
 	text))))
 
